src/redux/actions/experimentSettings/completeProcessingStep.js

```diff
@@ -17,23 +24,53 @@ const completeProcessingStep = (
     stepsDone: arrayStepsDone,
   };
 
-  await fetch(
-    `${getApiEndpoint()}/v1/experiments/${experimentId}/processingConfig`,
-    {
-      method: 'PUT',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify([{
-        name: 'meta',
-        body,
-      }]),
-    },
-  );
-
-  dispatch({
-    type: EXPERIMENT_SETTINGS_PROCESSING_COMPLETE_STEP,
-    payload:
-      { experimentId, settingName, numSteps },
-  });
+  try {
+    const response = await fetch(
+      `${getApiEndpoint()}/v1/experiments/${experimentId}/processingConfig`,
+      {
+        method: 'PUT',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify([{
+          name: 'meta',
+          body,
+        }]),
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error('HTTP status code was not 200.');
+    }
+
+    // If we had previously shown an error message, then now show a success one
+    if (completingStepError) {
+      dispatch(
+        pushNotificationMessage(
+          'success',
+          'Progress saved!',
+          3,
+        ),
+      );
+    }
+
+    dispatch({
+      type: EXPERIMENT_SETTINGS_PROCESSING_COMPLETE_STEP,
+      payload:
+        { experimentId, settingName, numSteps },
+    });
+  } catch (e) {
+    dispatch(
+      pushNotificationMessage(
+        'error',
+        'We couldn\'t connect to the server to save the current progress, please try again',
+        3,
+      ),
```

Comment:
Similarly, we can just say:

We couldn't save your processing settings. Please check your internet connection and try again.